### PR TITLE
Add remove-image command to pulp_container

### PR DESCRIPTION
**Pull request by gerrod3**
fixes: #566

Please be sure you have read our documentation on code contributions:
https://docs.pulpproject.org/pulp_cli/contributing/#code-conventions

### CODE REVIEW

File: pulpcore/cli/container/repository.py [hunk 1]
```diff
@@ -286,3 +287,21 @@ def copy_manifest(
...
+@pass_repository_context
+def remove_image(
+    repository_ctx: PulpContainerPushRepositoryContext,
+    digest: str,
+) -> None:
+    digest = digest.strip()
+    if not digest.startswith("sha256:"):
+        digest = f"sha256:{digest}"
+    if len(digest) != 71:  # len("sha256:") + 64
+        raise click.ClickException("Improper SHA256, please provide a valid 64 digit digest.")
```
Comment (mdellweg) [inline pulpcore/cli/container/repository.py:305]: Hmm, maybe not worth the trouble, but this looks like things we usually put in the option callback.

**Comment by gerrod3** [comment]
Filed a related issue for why the CI is currently not passing: https://github.com/pulp/pulp-cli/issues/580

File: pulpcore/cli/container/repository.py [hunk 2]
```diff
@@ -286,3 +287,21 @@ def copy_manifest(
         digests=digests or None,
         media_types=media_types or None,
     )
+
+
+@repository.command(allowed_with_contexts=push_container_context)
```
Comment (lubosmj) [inline pulpcore/cli/container/repository.py:292]: Does the `allowed_with_contexts` parameter have an impact on calling remove_image for regular container repositories?

Comment (gerrod3) [reply pulpcore/cli/container/repository.py:292]: Yes, the command doesn't show up if the context is regular container repositories. I don't see a remove_image call for regular container repositories.

**Review by lubosmj — APPROVED** [review]
I have tested the changes and it works as expected! It would be worth adding a simple test for the feature as well.

**Pull request closed without merging**